gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -511,88 +514,76 @@ function GM:KeyPress(ply, key)
 end
 
 ---
--- Runs when a IN key was released by a player.
--- For a more general purpose @{function} that handles all kinds of input, see @{GM:PlayerButtonUp}
--- @predicted
--- @param Player ply The @{Player} pressing the key. If running client-side, this will always be @{LocalPlayer}
--- @param number key The key that the @{Player} pressed using <a href="https://wiki.facepunch.com/gmod/Enums/IN">IN_Enums</a>.
--- @hook
--- @realm server
--- @ref https://wiki.facepunch.com/gmod/GM:KeyRelease
--- @local
-function GM:KeyRelease(ply, key)
-    if key ~= IN_USE or not IsValid(ply) or not ply:IsTerror() then
-        return
-    end
-
-    -- see if we need to do some custom usekey overriding
-    local tr = util.TraceLine({
-        start = ply:GetShootPos(),
-        endpos = ply:GetShootPos() + ply:GetAimVector() * 100,
-        filter = ply,
-        mask = MASK_SHOT,
-    })
+-- Normally all dead players are blocked from "+use" on the server, meaning we
+-- can't let them search bodies. This sucks because searching bodies is
+-- fun. Hence on the client we override +use for specs and use this instead.
+local function SpecUseKey(ply, ent)
+    -- Do not allow the spectator to gather information if they're about to revive.
+    if ply:IsReviving() then
+        LANG.Msg(ply, "spec_about_to_revive", nil, MSG_MSTACK_WARN)
 
-    if not tr.Hit or not IsValid(tr.Entity) then
         return
     end
 
-    if tr.Entity.CanUseKey and tr.Entity.UseOverride then
-        local phys = tr.Entity:GetPhysicsObject()
-
-        if IsValid(phys) and not phys:HasGameFlag(FVPHYSICS_PLAYER_HELD) then
-            tr.Entity:UseOverride(ply)
-
-            return true
+    if ent.player_ragdoll then
+        if not ply:KeyDown(IN_WALK) then
+            CORPSE.ShowSearch(ply, ent)
         else
-            -- do nothing, can't +use held objects
-            return true
+            ply:Spectate(OBS_MODE_IN_EYE)
+            ply:SpectateEntity(ent)
```

Comment:
I wasn't aware that you could observe corpses, lol. Since you already simplify this system I think you could probably remove that feature? I don't see the point of it

Reply:
I think I wont remove or change any functionality in this PR.